TNSAIonMatterInteractor.py

```diff
@@ -129,3 +97,65 @@ def saveData(self):
 
         series.flush()
         del series
+
+    def backengine(self):
+        pass
+
+    def run(self):
+        [mom, weight]= self.readSDF()
+        energy = np.square(mom)/(2*mp*e)
+        de=self.parameters.energy_bin
+        nb=math.floor(np.max(energy)/de)
+        counts = [0 for i in range(nb+1)]
+        binedges = [(i+1)*de for i in range(nb+1)]
+        print("Number of energy bins: %s" % len(binedges))
+        for en in energy:
+            index=math.floor(en/de)
+            counts[index] += weight[index]
+        [E, xs] = self.read_xsec()
+        xs = np.interp(binedges, E * 1.e6, xs * 1.e-28)
+
+        vx=[0]
+        for i in range(len(binedges)):
+            px = np.sqrt(2 * mp * binedges[i] * e)
+            Nd = counts[i]* self.parameters.ibeam_radius**2 / self.parameters.neutron_weight
+            inc = int(round(Nd * self.parameters.target_density * self.parameters.target_length * xs[i]))
```

Comment:
Is there a reason for using both int and round?

Reply:
As I remember, round() gave me float type number , but I need int.  I check it again